samples/HelloWorld/App.fs

```diff
@@ -1,83 +1,122 @@
 namespace HelloWorld
 
+open System
 open Fabulous
+open Fabulous.ScalarAttributeDefinitions
 open Fabulous.Maui
-open Microsoft.Maui
-open Microsoft.Maui.Graphics
-open Microsoft.Maui.Accessibility
-open Microsoft.Maui.Primitives
 
-open type Fabulous.Maui.View
+module ButtonExt =
+    let Clicked': ScalarAttributeDefinition<(unit -> unit), (unit -> unit)> =
+        let name = "Button_Clicked'"
+                    
+        let key =
+            SimpleScalarAttributeDefinition.CreateAttributeData(
+                ScalarAttributeComparers.noCompare,
+                (fun _ (newValueOpt: (unit -> unit) voption) node ->
+                    let event = (node.Target :?> Microsoft.Maui.Controls.Button).Clicked
 
-module App =
-    type Model = { Count: int; EnteredText: string }
+                    match node.TryGetHandler(name) with
+                    | ValueNone -> ()
+                    | ValueSome handler -> event.RemoveHandler handler
 
-    type Msg =
-        | Clicked
-        | UserWroteSomething of string
+                    match newValueOpt with
+                    | ValueNone -> node.SetHandler(name, ValueNone)
 
-    type CmdMsg = SemanticAnnounce of string
+                    | ValueSome(fn) ->
+                        let handler = EventHandler(fun _ _ -> fn())
 
-    let semanticAnnounce text =
-        Cmd.ofSub(fun _ -> SemanticScreenReader.Announce(text))
+                        event.AddHandler handler
+                        node.SetHandler(name, ValueSome handler))
+            )
 
-    let mapCmd cmdMsg =
-        match cmdMsg with
-        | SemanticAnnounce text -> semanticAnnounce text
+            |> AttributeDefinitionStore.registerScalar
 
-    let init () = { Count = 0; EnteredText = "" }, []
+        { Key = key; Name = name }
 
-    let update msg model =
-        match msg with
-        | Clicked -> { model with Count = model.Count + 1 }, [ SemanticAnnounce $"Clicked {model.Count} times" ]
+[<AutoOpen>]
+module ButtonBuildersExt =
+    type Fabulous.Maui.View with
+        static member inline Button'<'msg>(text: string, onClick: unit -> unit) =
+            WidgetBuilder<'msg, IFabButton>(
+                Button.WidgetKey,
+                Button.Text.WithValue(text),
+                ButtonExt.Clicked'.WithValue(onClick)
+            )
+        
+open type Fabulous.Maui.View
 
-        | UserWroteSomething text -> { model with EnteredText = text }, []
+module Counter =
+    let body =
+        view {
+            let! count = state 0
+            
+            VStack() {
+                Label($"Count is {count.Current}")
+                    .centerHorizontal()
+                
+                Button'("Increment", fun () -> count.Set(count.Current + 1))
+                Button'("Decrement", fun () -> count.Set(count.Current - 1))
```

Comment:
Any thoughts on supporting SingleChildBuilder. Aka similar to SwiftUI

```fsharp
Button("Decrement") {
     count.Set(count.Current - 1)
}
```

Reply:
Technically possible. Is it worth to break typical F# syntax for it though 🤔

Reply:
In SwiftUI, it's using the trailing closure syntax which is part of the language itself and widely used.
Here, we would make an exception just for it in F#.

Reply:
You are right. No worth faking `trailing closure syntax` just for this case